### PR TITLE
Tests fix

### PR DESCRIPTION
**Pull request by ahmedmuhsin**
- Run emulated tests instead of end-to-end tests in the official pipeline
- Modify stage dependencies and increases max parallel jobs to improve pipeline performance

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

**Review by hallvictoria — APPROVED** [review]

File: eng/ci/templates/jobs/run-emulated-tests-windows.yml [hunk 1]
```diff
@@ -1,9 +1,12 @@
+parameters:
+  poolName: ''
+
 jobs:
   - job: "TestWindows"
     displayName: 'Test java worker - Windows'
 
     pool:
-      name: 1es-pool-azfunc-public
+      name: ${{ parameters.poolName }}
```
Comment (manvkaur) [inline eng/ci/templates/jobs/run-emulated-tests-windows.yml:9]: this had issue with being parameterised, I remember I tried this out for some other pipeline, we can skip setting it in the job may be and it should get picked from the stage?

File: eng/ci/official-build.yml [hunk 2]
```diff
@@ -53,10 +53,13 @@ extends:
       - stage: TestWindows
         dependsOn: Build
         jobs:
-          - template: /eng/ci/templates/official/jobs/run-e2e-tests-windows.yml@self
```
Comment (gavin-aguiar) [inline eng/ci/official-build.yml:56]: Is run-e2e-tests-windows.yml being used anywhere else or can it be deleted?

File: eng/ci/templates/jobs/run-emulated-tests-windows.yml (hunk 1 above)
Comment (manvkaur) [reply eng/ci/templates/jobs/run-emulated-tests-windows.yml:9]: looks like it worked for the public pipeline, nvm.

File: eng/ci/official-build.yml (hunk 2 above)
Comment (ahmedmuhsin) [reply eng/ci/official-build.yml:56]: It is not being used anywhere currently but we might use it again in the future. I'll first need to determine how to migrate the rest of the e2e tests to emulated tests. If successful, I'll go ahead and remove the template files.

**Pull request merged**